sdk/turing/ensembler.py

```diff
@@ -279,6 +280,24 @@ def update(
             with the model. This will be passed to turing.ensembler.PyFunc.initialize().
             Example: {"config" : "config/staging.yaml"}
         """
+
+        relatedRouterVer = turing.Router.get_router_version_for_ensembler(
+            ensembler_id=self._id, status=[RouterStatus.PENDING]
+        )
+        if len(relatedRouterVer) > 0:
```

Comment:
> Do we need to duplicate these checks here? Won't the API do them anyway and return the same / similar error?

On this, I think the motivation was to prevent the creation of a new mlflow run in the first place (done in the SDK) if we shouldn't be updating the ensembler in the first place (if it's used by a router or a ensembling job). If we were to let the API handle the error, the new mlflow run would have already been created (though the API would rightfully delete the new run if it finds that the ensembler is being used a router or ensembling job). On hindsight, I think maybe we don't need so much redundancy in the code; we can just keep the code simple and rely on the API server to do the heavy lifting and checks.

Reply:
> Do we need to duplicate these checks here? Won't the API do them anyway and return the same / similar error?

I move the check from the api to the SDK, the update API doesn't have any more checking, as Zi Yi saying, the main purpose of putting the check in the SDK was to prevent the creation of a new mlflow run in the first place.

What do you guys think?, it is ok to let the SDK create the mlflow creation, since the API also clean it up (meaning that the check will be moved to the API instead of SDK). My idea at first was just to prevent more action (creation and deletion of the mlflow run)

Reply:
Ok, thanks for the clarification. Makes sense to prevent creation via the SDK's checks. Let's keep it this way.